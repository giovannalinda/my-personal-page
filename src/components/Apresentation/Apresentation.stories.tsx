import { Meta, Story } from '@storybook/react'
import { Apresentation } from './Apresentation'

export default {
  title: 'Components/Apresentation',
  component: Apresentation,
} as Meta

export const Default: Story = () => <Apresentation />
