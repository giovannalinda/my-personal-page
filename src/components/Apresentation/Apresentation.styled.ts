import styled from 'styled-components'
import { theme } from '~/styles'

export const Container = styled.section`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  align-items: center;
  margin-top: 26rem;

  @media screen and (max-width: ${theme.breakpoints.smallMobile}) {
    margin-top: 18rem;
  }
`

export const Content = styled.div`
  display: flex;
  text-align: left;
  flex-direction: column;

  @media screen and (max-width: ${theme.breakpoints.smallMobile}) {
    text-align: center;
  }
`

export const Title = styled.h1`
  font-size: ${theme.font.sizes.title};
  font-weight: ${theme.font.weigths.regular};
  max-width: 68rem;

  @media screen and (max-width: ${theme.breakpoints.smallDesktop}) {
    font-size: ${theme.font.sizes.big};
    padding-left: 10rem;
    max-width: 94rem;
  }

  @media screen and (max-width: ${theme.breakpoints.mobile}) {
    font-size: ${theme.font.sizes.title};
    margin-top: 18rem;
    text-align: center;
  }

  @media screen and (max-width: ${theme.breakpoints.smallMobile}) {
    padding-left: ${theme.space[7]};
    font-size: ${theme.font.sizes.subtitle};
  }
`

export const Paragraph = styled.p`
  color: ${theme.colors.gray400};
  padding-top: ${theme.space[7]};
  font-size: ${theme.font.sizes.verySmall};

  @media screen and (max-width: ${theme.breakpoints.smallDesktop}) {
    font-size: ${theme.font.sizes.paragraph};
    padding-left: 10rem;
  }

  @media screen and (max-width: ${theme.breakpoints.smallMobile}) {
    padding-left: 4rem;
  }
`

export const Link = styled.a`
  color: ${theme.colors.white};
  font-size: ${theme.font.sizes.subtitle};
  text-decoration: underline;
  padding-top: ${theme.space[2]};

  @media screen and (max-width: ${theme.breakpoints.smallDesktop}) {
    font-size: ${theme.font.sizes.subtitle};
    padding-left: 10rem;
  }

  @media screen and (max-width: ${theme.breakpoints.smallMobile}) {
    padding-left: 4rem;
  }
`

export const AboutImageWrapper = styled.div`
  position: absolute;
  right: 0;
  padding-right: 20rem;

  @media screen and (max-width: ${theme.breakpoints.smallDesktop}) {
    display: none;
  }
`
