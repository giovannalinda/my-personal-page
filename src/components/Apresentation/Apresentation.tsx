import { motion } from 'framer-motion'
import Image from 'next/image'
import { Assets } from '~/assets'

import * as S from './Apresentation.styled'

const TITLE = 'Construindo produtos, marcas e experiÃªncias digitais.'
const PARAGRAPH = 'ENTRE EM CONTATO POR AQUI'
const EMAIL = 'eugiovannasouza@gmail.com'

const letterVariant = {
  hidden: { opacity: 0 },
  visible: { opacity: 1 },
}

export function Apresentation() {
  return (
    <S.Container>
      <S.Content>
        <S.Title>
          {TITLE.split('').map((char, index) => (
            <motion.span
              key={`${char}-${index}`}
              variants={letterVariant}
              transition={{
                delay: index / TITLE.length,
                duration: 0.4,
              }}
              initial='hidden'
              animate='visible'
            >
              {char}
            </motion.span>
          ))}
        </S.Title>
        <S.Paragraph>
          {PARAGRAPH.split('').map((char, index) => (
            <motion.span
              key={`${char}-${index}`}
              variants={letterVariant}
              transition={{
                delay: index / PARAGRAPH.length,
                duration: 0.4,
              }}
              initial='hidden'
              animate='visible'
            >
              {char}
            </motion.span>
          ))}
        </S.Paragraph>
        <S.Link href='mailto:eugiovannasouza@gmail.com'>
          {EMAIL.split('').map((char, index) => (
            <motion.span
              key={`${char}-${index}`}
              variants={letterVariant}
              transition={{
                delay: index / PARAGRAPH.length,
                duration: 0.4,
              }}
              initial='hidden'
              animate='visible'
            >
              {char}
            </motion.span>
          ))}
        </S.Link>
      </S.Content>

      <S.AboutImageWrapper>
        <Image
          src={Assets.ApresentationImg}
          width={300}
          height={600}
          objectFit='cover'
        />
      </S.AboutImageWrapper>
    </S.Container>
  )
}
