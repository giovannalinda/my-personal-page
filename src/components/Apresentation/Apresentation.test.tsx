import { render } from '~/__helpers__/app-tests'
import { Apresentation } from './Apresentation'

describe('Apresentation', () => {
  it('matches snapshot', () => {
    const { container } = render(<Apresentation />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
