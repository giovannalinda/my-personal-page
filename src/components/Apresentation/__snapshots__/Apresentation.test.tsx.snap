// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apresentation matches snapshot 1`] = `
<section
  class="sc-bczRLJ empGxY"
>
  <div
    class="sc-gsnTZi jwBwLH"
  >
    <h1
      class="sc-dkzDqf kCGsYS"
    >
      <span
        style="opacity: 0;"
      >
        C
      </span>
      <span
        style="opacity: 0;"
      >
        o
      </span>
      <span
        style="opacity: 0;"
      >
        n
      </span>
      <span
        style="opacity: 0;"
      >
        s
      </span>
      <span
        style="opacity: 0;"
      >
        t
      </span>
      <span
        style="opacity: 0;"
      >
        r
      </span>
      <span
        style="opacity: 0;"
      >
        u
      </span>
      <span
        style="opacity: 0;"
      >
        i
      </span>
      <span
        style="opacity: 0;"
      >
        n
      </span>
      <span
        style="opacity: 0;"
      >
        d
      </span>
      <span
        style="opacity: 0;"
      >
        o
      </span>
      <span
        style="opacity: 0;"
      >
         
      </span>
      <span
        style="opacity: 0;"
      >
        p
      </span>
      <span
        style="opacity: 0;"
      >
        r
      </span>
      <span
        style="opacity: 0;"
      >
        o
      </span>
      <span
        style="opacity: 0;"
      >
        d
      </span>
      <span
        style="opacity: 0;"
      >
        u
      </span>
      <span
        style="opacity: 0;"
      >
        t
      </span>
      <span
        style="opacity: 0;"
      >
        o
      </span>
      <span
        style="opacity: 0;"
      >
        s
      </span>
      <span
        style="opacity: 0;"
      >
        ,
      </span>
      <span
        style="opacity: 0;"
      >
         
      </span>
      <span
        style="opacity: 0;"
      >
        m
      </span>
      <span
        style="opacity: 0;"
      >
        a
      </span>
      <span
        style="opacity: 0;"
      >
        r
      </span>
      <span
        style="opacity: 0;"
      >
        c
      </span>
      <span
        style="opacity: 0;"
      >
        a
      </span>
      <span
        style="opacity: 0;"
      >
        s
      </span>
      <span
        style="opacity: 0;"
      >
         
      </span>
      <span
        style="opacity: 0;"
      >
        e
      </span>
      <span
        style="opacity: 0;"
      >
         
      </span>
      <span
        style="opacity: 0;"
      >
        e
      </span>
      <span
        style="opacity: 0;"
      >
        x
      </span>
      <span
        style="opacity: 0;"
      >
        p
      </span>
      <span
        style="opacity: 0;"
      >
        e
      </span>
      <span
        style="opacity: 0;"
      >
        r
      </span>
      <span
        style="opacity: 0;"
      >
        i
      </span>
      <span
        style="opacity: 0;"
      >
        Ãª
      </span>
      <span
        style="opacity: 0;"
      >
        n
      </span>
      <span
        style="opacity: 0;"
      >
        c
      </span>
      <span
        style="opacity: 0;"
      >
        i
      </span>
      <span
        style="opacity: 0;"
      >
        a
      </span>
      <span
        style="opacity: 0;"
      >
        s
      </span>
      <span
        style="opacity: 0;"
      >
         
      </span>
      <span
        style="opacity: 0;"
      >
        d
      </span>
      <span
        style="opacity: 0;"
      >
        i
      </span>
      <span
        style="opacity: 0;"
      >
        g
      </span>
      <span
        style="opacity: 0;"
      >
        i
      </span>
      <span
        style="opacity: 0;"
      >
        t
      </span>
      <span
        style="opacity: 0;"
      >
        a
      </span>
      <span
        style="opacity: 0;"
      >
        i
      </span>
      <span
        style="opacity: 0;"
      >
        s
      </span>
      <span
        style="opacity: 0;"
      >
        .
      </span>
    </h1>
    <p
      class="sc-hKMtZM fAtVar"
    >
      <span
        style="opacity: 0;"
      >
        E
      </span>
      <span
        style="opacity: 0;"
      >
        N
      </span>
      <span
        style="opacity: 0;"
      >
        T
      </span>
      <span
        style="opacity: 0;"
      >
        R
      </span>
      <span
        style="opacity: 0;"
      >
        E
      </span>
      <span
        style="opacity: 0;"
      >
         
      </span>
      <span
        style="opacity: 0;"
      >
        E
      </span>
      <span
        style="opacity: 0;"
      >
        M
      </span>
      <span
        style="opacity: 0;"
      >
         
      </span>
      <span
        style="opacity: 0;"
      >
        C
      </span>
      <span
        style="opacity: 0;"
      >
        O
      </span>
      <span
        style="opacity: 0;"
      >
        N
      </span>
      <span
        style="opacity: 0;"
      >
        T
      </span>
      <span
        style="opacity: 0;"
      >
        A
      </span>
      <span
        style="opacity: 0;"
      >
        T
      </span>
      <span
        style="opacity: 0;"
      >
        O
      </span>
      <span
        style="opacity: 0;"
      >
         
      </span>
      <span
        style="opacity: 0;"
      >
        P
      </span>
      <span
        style="opacity: 0;"
      >
        O
      </span>
      <span
        style="opacity: 0;"
      >
        R
      </span>
      <span
        style="opacity: 0;"
      >
         
      </span>
      <span
        style="opacity: 0;"
      >
        A
      </span>
      <span
        style="opacity: 0;"
      >
        Q
      </span>
      <span
        style="opacity: 0;"
      >
        U
      </span>
      <span
        style="opacity: 0;"
      >
        I
      </span>
    </p>
    <a
      class="sc-eCYdqJ cmOTFM"
      href="mailto:eugiovannasouza@gmail.com"
    >
      <span
        style="opacity: 0;"
      >
        e
      </span>
      <span
        style="opacity: 0;"
      >
        u
      </span>
      <span
        style="opacity: 0;"
      >
        g
      </span>
      <span
        style="opacity: 0;"
      >
        i
      </span>
      <span
        style="opacity: 0;"
      >
        o
      </span>
      <span
        style="opacity: 0;"
      >
        v
      </span>
      <span
        style="opacity: 0;"
      >
        a
      </span>
      <span
        style="opacity: 0;"
      >
        n
      </span>
      <span
        style="opacity: 0;"
      >
        n
      </span>
      <span
        style="opacity: 0;"
      >
        a
      </span>
      <span
        style="opacity: 0;"
      >
        s
      </span>
      <span
        style="opacity: 0;"
      >
        o
      </span>
      <span
        style="opacity: 0;"
      >
        u
      </span>
      <span
        style="opacity: 0;"
      >
        z
      </span>
      <span
        style="opacity: 0;"
      >
        a
      </span>
      <span
        style="opacity: 0;"
      >
        @
      </span>
      <span
        style="opacity: 0;"
      >
        g
      </span>
      <span
        style="opacity: 0;"
      >
        m
      </span>
      <span
        style="opacity: 0;"
      >
        a
      </span>
      <span
        style="opacity: 0;"
      >
        i
      </span>
      <span
        style="opacity: 0;"
      >
        l
      </span>
      <span
        style="opacity: 0;"
      >
        .
      </span>
      <span
        style="opacity: 0;"
      >
        c
      </span>
      <span
        style="opacity: 0;"
      >
        o
      </span>
      <span
        style="opacity: 0;"
      >
        m
      </span>
    </a>
  </div>
  <div
    class="sc-jSMfEi ykkjU"
  >
    <img
      height="600"
      src="/apresentation.svg"
      width="300"
    />
  </div>
</section>
`;
